### DISCUSSION TITLE
Open source

### DISCUSSION THREAD

**Discussion by Ankush11903** [community/community#43216] [New to GitHub] [answered]
Hey !
i am n newbie here.
so,just want to explore the field of open-source.
but don't know how to get started.
so,it would be great if someone can help me to get out of this delima. :)
happy coding.

**Comment by LuisContrerasGlz** [comment]
Some things that you can do are:

Find a project that you are interested in and start contributing to it. Many open-source projects have a list of "good first issues" that are suitable for new contributors.

Join an open-source community (GitHub, Reddit, and Stack Exchange.)

Start your own open-source project. 

Learn about open-source licenses and how to use them. Understanding open-source licenses will help you ensure that you are using open-source software legally and appropriately.

Attend open-source events or conferences. There are many conferences and meetups that focus on open-source, and these can be a great way to learn more and meet other people in the community.

**Comment by diva-21** [comment, marked as answer]
Have a good and personalized README.md
Visit [Eddie Jaoude course ](https://www.eddiejaoude.io/course-github-profile-landing)for README customization.

On youtube, follow Kunal Kushwaha , Eddie and other Open Source guys at Twitter.
Most of your answers are on Twitter daily
Join communities, start looking issues where ever your understanding is fitting

Try to get badges
Use emojis to clarify tone. Compare “Looks good :)” or “Looks good! 👍 ” with “It’s fine.”
Commit small changes
Good commit messages describing what the commit is for

Do it daily with consistency , and you will love the process.

To make PR, you can follow these steps:

NOTE: You cant override/change somebody's else code directly. So if you want to contribute to them i.e want to change and make the head of that repository guy, you have to keep a Pull Request with your changes.


So What is Pull Request (PR) ??
Its a request to the head of the repo that i made a few changes(added/modified/deleted) in your project ,so that the head will see those changes. If the changes are valid , then he will merge your pr. Else he adds some modifications to be done on your Pr.
**Steps to keep PR.**

Fork the main project. Fork makes a copy of that into your account.

In your repos, you can see the forked repo, so now you have to Clone that repo. Copy the url which you find on clicking the code (green button) on your forked repo.

Go to your desktop and open a terminal/git bash. Type :- git clone url . Then press enter.

Go to the project directory. Open a new termianal. Create a new branch by typing:
git checkout -b branch_name . example: git checkout -b diva
Then enter

Make the changes in the files in which you want.

Now press git status. You can see the files which have been modified in red color.

Now we have to add them. so type: git add .
NOTE: Its git add . ( ' . ' means adding all. Dont misunderstood that as a fullstop)

Now we have to make commit (meaning its like taking a photograph of the relatives standing on the stage) by typing:-
git commit -m "message" . Example: git commit -m "added modified files" . then enter

Now we have to push these modified files to your repo by typing:
git push origin your_branch_name . Example: git push origin diva.

Open the forked repo, you can see a pull request green button displayed on the screen. Click on that and you will be redirected to the head repository. Now Write the description of those changes in that dialog box and click on Pull Request.

Huraah , you made your pr successfully 🥳🥳🥳

To learn more about GitHub and Open Source, you can follow me and ask.
And to learn github basics, watch this tutorial by [Kunal Kushwaha](https://youtu.be/apGV9Kg7ics)

All the best.
Thank you

**Reply by Ankush11903** [reply]
Thanks for the Guidance.
Can you elaborate on how to find Repo for pr?

**Reply by diva-21** [reply]
First of all choose those domains in which you have some idea.
Ex: If u know DSA, then search Data Structures & Algos , you will find many.
Or if u did Web Dev, then search it directly, and start keeping Prs.
Even you can start just updating ReadMe files. 

You should join communities like EddieHub,etc. There you can find issues to solve.

You can follow me, i have made many prs on Web dev and dsa.
Thank you

**Reply by YahefuYusufu** [reply]
hi guys ,I need open project also(web dev) , if someone interested can do together also.

**Reply by Ankush11903** [reply]
Hello @YusufYakup.
Yes, I am Interested.

**Reply by YahefuYusufu** [reply]
do you have any idea or some project? 
we can do it together ?

**Reply by Harmonygrams** [reply]
I'm interested too. I'm a frontend React Developer. Wouldn't mind working for free
